2016/proposals/deng-pan.md

```diff
@@ -0,0 +1,130 @@
+#Integrating `pandas.Panel` and xarray Features
+
+## Abstract
+
+Pandas package excels at processing tabular models, especially 1-D and 2-D models with data structure `pd.Series` and `pd.DataFrame`. However, despite the specialized data structure for 3-D and N-D data, `pd.Panel`, pandas is not well-designed for high dimensional data processing. Xarray was then developed to compensate the weak part of pandas. Xarray package enabled statistic and mathematical analysis of high-dimensional data, and implemented several new features that are missing in pandas.
+ 
+However, some useful features in pandas were not implemented in xarray yet. Thus, my project will be focusing on porting features from pandas to xarray, and improving data structure conversion between pandas and xarray. By the end of my work, I expect pandas/xarray users can have specialized tools for different types of data processing, and migrate between pandas and xarray smoothly. 
+
+## Technical Details
+
+Most of my proposal is supposed to be carried out with current implemented features in pandas and xarray. For PCA part, to improve the performance, I might switch to C++ and Eigen3 library.
+ 
+
+## Schedule of Deliverables
+
+###Milestone 1: Migrate features from pandas to xarray 
+
+**Week 1 - 5, May 23rd - June 26th**
+
+  - I will have my coursework finished within the first two weeks
+  - I will implement more multiIndex-related features and port features from pandas to xarray.
+
+    - **Week 1:** Enable multi-datatype in `xr.DataArray` like `pd.panel`
+    - **Week 2:** Enable groupby for one-dimensional variables / Make levels accessible as coordinate variables
+    - **Week 3:** Based on [#702](https://github.com/pydata/xarray/pull/702) and according to [#719](https://github.com/pydata/xarray/issues/719), I will implement selection return objects with MultiIndex, and add `set_index`/`reset_index`/`swaplevel` to make it easier to create and manipulate multi-indexes.
+    - **Week 4 - 5:** Port features from pd.panel to `xr.DataArray`, such as `prod`, `cumsum`, `rank`, and etc.. Ref [#791](https://github.com/pydata/xarray/issues/791)
+
+###Milestone 2: Type conversion between pandas and xarray data structure
+
+**Week 6 - 9, June 27th - July 24th**
+
+  - To ensure that data can migrate between pandas and xarray data structure smoothly, I want to spend time on optimizing the conversion between data types. I will check current conversion methods, and implement more features - from multi-indexed/ hierarchical-indexed `pd.DataFrame` to `xr.DataSet`; from xr.DataSet to 2-D pd.DataFrame with PCA.
+
+    - **Week 6:** Test current conversion methods with newly added features in Milestone1 and fix existing bugs.
+    - **Week 7:** Based on [#702](https://github.com/pydata/xarray/pull/702) and Milestone1, I am going to take one step further and implement conversion from multiIndexed `pd.DataFrame` to `xr.DataSet`.
```

Comment:
This actually already works in the latest release of xarray -- is there an outstanding issue about this you found?